CefSharp/DependencyChecker.cs

```diff
@@ -81,64 +91,44 @@ public static class DependencyChecker
-        public static List<string> CheckDependencies(bool checkOptional, bool packLoadingDisabled, string path, string resourcesDirPath, string localePackFile = LocalesPackFile)
+        public static List<string> CheckDependencies(bool checkOptional, bool packLoadingDisabled, string path, string resourcesDirPath, string browserSubProcessPath, string localePackFile = LocalesPackFile)
         {
             var missingDependencies = new List<string>();
 
-            //Loop through Cef dependencies and add to list if not found
-            foreach (var cefDependency in CefDependencies)
-            {
-                var dependencyPath = Path.Combine(path, cefDependency);
-
-                if (!File.Exists(dependencyPath))
-                {
-                    missingDependencies.Add(cefDependency);
-                }
-            }
+            missingDependencies.AddRange(CheckDependencyList(path, CefDependencies));
 
             if (!packLoadingDisabled)
             {
-                //Loop through Cef Resources and add to list if not found
-                foreach (var cefResource in CefResources)
-                {
-                    var resourcePath = Path.Combine(resourcesDirPath, cefResource);
-
-                    if (!File.Exists(resourcePath))
-                    {
-                        missingDependencies.Add(cefResource);
-                    }
-                }
+                missingDependencies.AddRange(CheckDependencyList(resourcesDirPath, CefResources));
             }
 
             if (checkOptional)
             {
-                //Loop through Cef Optional dependencies and add to list if not found
-                foreach (var cefDependency in CefOptionalDependencies)
-                {
-                    var dependencyPath = Path.Combine(path, cefDependency);
-
-                    if (!File.Exists(dependencyPath))
-                    {
-                        missingDependencies.Add(cefDependency);
-                    }
-                }
+                missingDependencies.AddRange(CheckDependencyList(path, CefOptionalDependencies));
             }
 
-            // Loop through CefSharp dependencies and add to list if not found
-            foreach (var cefSharpDependency in CefSharpDependencies)
+            missingDependencies.AddRange(CheckDependencyList(path, CefSharpDependencies));
+
+            if (!File.Exists(browserSubProcessPath))
             {
-                var dependencyPath = Path.Combine(path, cefSharpDependency);
+                missingDependencies.Add(browserSubProcessPath);
+            }
 
-                if (!File.Exists(dependencyPath))
-                {
-                    missingDependencies.Add(cefSharpDependency);
-                }
+            var browserSubprocessDir = Path.GetDirectoryName(browserSubProcessPath);
+            if (browserSubprocessDir == null)
+            {
+                missingDependencies.AddRange(BrowserSubprocessDependencies);
```

Comment:
No at this point browserSubProcessPath is something that doesn't exist, so everything that should be in there is by default missing.  I considered adding browserSubProcessPath itself to the collection instead, but that didn't seem consistent with the rest of the code.  Also browserSubProcessPath could be null and then there's not enough explanation of what's wrong.